lib/schnorr.py

```
@@ -0,0 +1,101 @@
+        library_paths = 'libsecp256k1.so',
+    else:
+        library_paths = (os.path.join(os.path.dirname(__file__), 'libsecp256k1.so.0'),  # on linux we install it alongside the python scripts.
+                         'libsecp256k1.so.0')  # fall back to system lib, if any
+
+    for lp in library_paths:
+        try:
+            secp256k1 = ctypes.cdll.LoadLibrary(lp)
+        except:
+            continue
+        if secp256k1:
+            break
+    if not secp256k1:
+        print_stderr('[schnorr] warning: libsecp256k1 library failed to load')
+        return None
+
+    try:
+        secp256k1.secp256k1_context_create.argtypes = [c_uint]
+        secp256k1.secp256k1_context_create.restype = c_void_p 
+
+        secp256k1.secp256k1_context_randomize.argtypes = [c_void_p, c_char_p]
+        secp256k1.secp256k1_context_randomize.restype = c_int
+
+        secp256k1.secp256k1_schnorr_sign.argtypes = [ c_void_p, c_char_p, c_char_p, c_char_p, c_char_p, c_char_p]
+        secp256k1.secp256k1_schnorr_sign.restype = c_int 
+
+        secp256k1.ctx = secp256k1.secp256k1_context_create(SECP256K1_CONTEXT_SIGN | SECP256K1_CONTEXT_VERIFY)
+        r = secp256k1.secp256k1_context_randomize(secp256k1.ctx, os.urandom(32))
+        if r:
+            return secp256k1
+        else:
+            print_stderr('[schnorr] warning: secp256k1_context_randomize failed')
+            return None
+    except (OSError, AttributeError):
+        #traceback.print_exc(file=sys.stderr)
+        print_stderr('[schnorr] warning: libsecp256k1 library was found and loaded but there was an error when using it')
+        return None 
+
+try:
+    _libsecp256k1 = load_library()
+except:
+    _libsecp256k1 = None
+    #traceback.print_exc(file=sys.stderr)
+
+def sign_schnorr(message,privkey):
+    
+    # Error handling in case of libsec failure:
+    wefailed= False 
+    try:
+        _libsecp256k1 = load_library()
+    except:
+        _libsecp256k1 = None
+        wefailed= True
```

You should just `return bytes((0x0,))` rather than setting `wefailed= True`.  However, you really shouldn't be doing either.  Just let the exception bubble up unless there is a really good reason not to.  In which case, return a tuple. (e.g. `return (False, bytes((0x0,)),)`.

Then your call can explicitly check.